### DISCUSSION TITLE
只有网页端风控，APP不风控

### DISCUSSION THREAD

**Discussion by Z-fly** [lanyeeee/bilibili-manga-downloader#36] [General]
![image](https://github.com/user-attachments/assets/becf3b47-f782-4a62-ae9d-106bc63ee145)

**Comment by lanyeeee** [comment]
这是什么意思，是APP的截图还是下载到的图片长这样？

**Comment by Z-fly** [comment]
![image](https://github.com/user-attachments/assets/3d5f535f-e40e-46ee-b460-2fc16f64f942)

**Comment by lanyeeee** [comment]
没有，我能正常下载，你这个情况是风控了吧
https://github.com/lanyeeee/bilibili-manga-downloader/issues/31#issuecomment-2480554818

**Comment by Z-fly** [comment]
但是app是正常看的 这是风控吗

**Comment by lanyeeee** [comment]
> ![image](https://private-user-images.githubusercontent.com/10470892/386853557-becf3b47-f782-4a62-ae9d-106bc63ee145.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3Njg3OTMsIm5iZiI6MTczMTc2ODQ5MywicGF0aCI6Ii8xMDQ3MDg5Mi8zODY4NTM1NTctYmVjZjNiNDctZjc4Mi00YTYyLWFlOWQtMTA2YmM2M2VlMTQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE2VDE0NDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI5NWVmMWZhNmJhYzY5NzdiZjU0NmIyMjQzYWJjOWZlOTQzNDMwNGQ5ODNkYmQ0YmFiNzdmZjE3YmQyYjY4YjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.h-MvgNNmkP8O41HjQ6HlXTV9xin3roe63OpnUu-TlyU)


你这个截图是哪来的

**Comment by lanyeeee** [comment]
你意思是说网页版没法看，只能app看？

**Comment by Z-fly** [comment]
是的

**Comment by lanyeeee** [comment]
没见过这种类型的风控。
我遇到的风控都是网页和app都不能看的，然后要去app短信验证解除(风控后随便点开一个漫画观看就弹窗了)
那边有让你短信验证过吗？

**Comment by Z-fly** [comment]
没有要求短信验证

**Comment by Z-fly** [comment]
![image](https://github.com/user-attachments/assets/f6f19617-0c65-4cb6-b15f-650922ab6019)

**Comment by lanyeeee** [comment]
这我也没法复现啊。。。
网页重新登录有用吗？网页免费章节能看吗？用无痕模式能看网页的免费章节吗？
例如https://manga.bilibili.com/mc26470/313073

**Comment by Z-fly** [comment]
网页都看不了 无痕模式在没登录的情况下可以看

**Comment by lanyeeee** [comment]
那就可以确定是账号的问题了
抓个包看看，这俩返回的code和msg是什么
![image](https://github.com/user-attachments/assets/4986aa75-0330-4234-bc7b-75173fca9719)

**Comment by lanyeeee** [comment]
![image](https://github.com/user-attachments/assets/d8f38b2b-119f-4ee2-8756-fdebdaf1c24b)

**Comment by Z-fly** [comment]
![image](https://github.com/user-attachments/assets/500314a3-27cd-4fac-a861-270e2fd3c152)
![image](https://github.com/user-attachments/assets/9b177669-fcb1-4467-9c86-3920c7d68ef7)

**Comment by lanyeeee** [comment]
不是GetHotSearch，是这个
![image](https://github.com/user-attachments/assets/a0ec6190-01b1-4155-a556-34d3b3a882d5)

**Comment by Z-fly** [comment]
![image](https://github.com/user-attachments/assets/f0498a80-3b42-4c3a-b4ec-8dbbc00e37ca)

**Comment by lanyeeee** [comment]
看起来能正常请求获取图片index的接口
但是从index转url的接口请求失败

你方便手动请求一次index转url的接口吗？
按照下面的步骤
# 1. 复制ImageToken请求为curl
![image](https://github.com/user-attachments/assets/170c277b-5cb8-49ce-a0dc-7747f10279d4)


# 2. 粘贴到postman中
![image](https://github.com/user-attachments/assets/d216f2d1-9fd8-475c-a263-b3d46b093524)


# 3. 复制GetImageIndex响应中`data.images[2].path`的值
![image](https://github.com/user-attachments/assets/2996b674-da0d-47dd-b142-4a4a9b092342)


# 4. 替换刚才postman中请求body中对应的位置
![image](https://github.com/user-attachments/assets/6958469a-0239-466b-a04e-6f9bafe5cdad)

# 5. 发送请求，把响应里的complete_url粘贴到浏览器里，看看还是不是那张风控的图片
![image](https://github.com/user-attachments/assets/1ac6b57b-0e69-4757-a75a-f0dca2049d2a)

如果访问complete_url**无法正常显示图片就说明正常**，如果显示风控的图片就不正常

**Comment by Z-fly** [comment]
直接cmd curl请求就行了
![image](https://github.com/user-attachments/assets/2726f097-1ea4-4718-a7b3-6a4ab54d3846)
这个被风控了说明
https://i0.hdslb.com/bfs/manga-static/eb864b6282a7ec0fd86821e22fd956b0798c6543.png

**Comment by Z-fly** [comment]
手机上能看，说明这个风控理论可以绕过

**Comment by lanyeeee** [comment]
那我就无能为力了
可以肯定是账号出了问题，无法正常请求ImageToken接口

**Comment by lanyeeee** [comment]
你用v0.6应该可以正常下载，它用的是app的接口，但是app接口用多了容易风控，所以v0.7又迁移回web接口了
如果你一定要用app接口进行下载，记得限制并发

**Comment by Z-fly** [comment]


**Reply by lanyeeee** [reply]
试试这个
[bilibili-manga-downloader.zip](https://github.com/user-attachments/files/17787145/bilibili-manga-downloader.zip)

**Reply by Z-fly** [reply]
测试过了 一样的还是只能下载风控图

**Reply by lanyeeee** [reply]
那我也没辙了，上面发的严格使用app的接口。难道你的账号跟普通的号请求不一样？
既然你app能看，你可以自己抓包看看app是怎么拿图片的，然后自己发包看看能不能正确拿到图片

**Reply by Z-fly** [reply]
我觉得应该是要模拟最新app请求手机端的漫画页面，然后再请求图片章节

**Reply by lanyeeee** [reply]
你先自己抓包看看可行性，我总不能随便就按你说的改，然后发现不行导致无效工作
